training/src/anemoi/training/diagnostics/mlflow/logger.py

```diff
@@ -391,20 +390,34 @@ def _get_mlflow_run_params(
             self.auth.authenticate()
             mlflow_client = mlflow.MlflowClient(tracking_uri)
 
-            if config_run_id and on_resume_create_child:
+            # This block is used when a run ID is specified with child runs option activated
+            if config_run_id and on_resume_create_child and not fork_run_id:
                 parent_run_id = config_run_id  # parent_run_id
                 parent_run = mlflow_client.get_run(parent_run_id)
                 run_name = parent_run.info.run_name
                 self._check_server2server_lineage(parent_run)
                 tags["mlflow.parentRunId"] = parent_run_id
                 tags["resumedRun"] = "True"  # tags can't take boolean values
-            elif config_run_id and not on_resume_create_child:
+            # This block is used when a run ID is specified without child runs option activated
+            elif config_run_id and not on_resume_create_child and not fork_run_id:
                 run_id = config_run_id
                 run = mlflow_client.get_run(run_id)
                 run_name = run.info.run_name
                 self._check_server2server_lineage(run)
                 mlflow_client.update_run(run_id=run_id, status="RUNNING")
                 tags["resumedRun"] = "True"
+            # This block is used when a run is forked and an existing run ID is specified
+            # Child run option is activated
+            elif config_run_id and fork_run_id:
```

Comment:
This is not really something that could happen. Either you decide to resume a run using the 'run_id' or you decide to fork using the 'fork_run_id'

Reply:
This has been discussed offline, it is needed if we want to propagate the dry run ID a the training that uses a forked run ID.